pkg/integration/event_integration.go

```diff
@@ -94,7 +94,7 @@ func NewEventCalendarDataFetcher(eventAPIEndpoint string) (*EventCalendarDataFet
 
 func (ec *EventCalendarDataFetcher) GetDesiredEvents(startTime time.Time, endTime time.Time) ([]EventDetails, error) {
 	var eventDetails []EventDetails
-	fetchEventsUrl := fmt.Sprintf("http://%s/fk-event-calendar-service/v1/eventCalendar/search?startTime=%d&tier=%s&pageNo=%d&pageSize=%d", ec.EventAPIEndpoint, startTime.UnixMilli(), "ALL_MP", 0, 100)
+	fetchEventsUrl := fmt.Sprintf("%s?startTime=%d&tier=%s&pageNo=%d&pageSize=%d", ec.EventAPIEndpoint, startTime.UnixMilli(), "ALL_MP", 0, 100)
```

Comment:
This still doesnt look generic. There are assumptions about URL pattern.

Reply:
Ignore. This is abstracted via an interface, so fk api specific implementation here should be ok.

Reply:
We are discussing about moving FK specific implementation into a separate repo. We will take that up later.